### PR TITLE
feat: rollback testcases

### PR DESCRIPTION
**Pull request by sdpisreddevil**
### Commit Message

```bash
type: commit message
```

see the [guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#commit-messages) for commit messages.

### Changelog Entry

```bash
version: <log entry>
```

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have documented my code in accordance with the [documentation guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#documentation)
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have run the unit tests
- [ ] I only have one commit (if not, squash them into one commit).
- [ ] I have a descriptive commit message that adheres to the [commit message guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#commit-messages)

> Please review the [CONTRIBUTING.md](/CONTRIBUTING.md) file for detailed contributing guidelines.

### CODE REVIEW

**Review by AntonAndell — CHANGES_REQUESTED** [review]
Test cases look good, but need to be merged

File: contracts/evm/xcall/test/CallService.t.sol [hunk 1]
```diff
@@ -190,4 +219,202 @@ contract CallServiceTest is Test {
...
+
+        uint256 sn = callService.sendCallMessage{value: 0 ether}(iconDapp, data, rollbackData, _baseSource, _baseDestination);
+        assertEq(sn, 1);
+
+        Types.CSMessageResponse memory response = Types.CSMessageResponse(1, Types.CS_RESP_SUCCESS);
+        Types.CSMessage memory msg = Types.CSMessage(Types.CS_RESPONSE, RLPEncodeStruct.encodeCSMessageResponse(response));
+
+        vm.expectEmit();
+        emit ResponseMessage(1, Types.CS_RESP_SUCCESS);
+        emit RollbackMessage(1);
```
[the remark this answers is not in the document]
Comment (sdpisreddevil) [reply contracts/evm/xcall/test/CallService.t.sol]: updated

File: contracts/evm/xcall/test/CallService.t.sol [hunk 2]
```diff
@@ -190,4 +218,201 @@ contract CallServiceTest is Test {
...
+
+        Types.CSMessageResponse memory response = Types.CSMessageResponse(1, Types.CS_RESP_FAILURE);
+        Types.CSMessage memory msg = Types.CSMessage(Types.CS_RESPONSE, RLPEncodeStruct.encodeCSMessageResponse(response));
+
+        vm.prank(address(connection1));
+        callService.handleMessage(iconNid, RLPEncodeStruct.encodeCSMessage(msg));
+
+        vm.expectEmit();
+        emit ResponseMessage(1, Types.CS_RESP_FAILURE);
+        emit RollbackMessage(1);
```
Comment (AntonAndell) [inline contracts/evm/xcall/test/CallService.t.sol:391]: Also fix emit here

File: contracts/evm/xcall/test/CallService.t.sol [hunk 3]
```diff
@@ -190,4 +218,201 @@ contract CallServiceTest is Test {
...
+
+        uint256 sn = callService.sendCallMessage{value: 0 ether}(iconDapp, data, rollbackData, _baseSource, _baseDestination);
+        assertEq(sn, 1);
+
+        Types.CSMessageResponse memory response = Types.CSMessageResponse(1, Types.CS_RESP_FAILURE);
+        Types.CSMessage memory msg = Types.CSMessage(Types.CS_RESPONSE, RLPEncodeStruct.encodeCSMessageResponse(response));
+
+        vm.expectEmit();
+        emit ResponseMessage(1, Types.CS_RESP_FAILURE);
+        emit RollbackMessage(1);
```
Comment (AntonAndell) [inline contracts/evm/xcall/test/CallService.t.sol:331]: :point_up:

File: contracts/evm/xcall/test/CallService.t.sol [hunk 4]
```diff
@@ -190,4 +218,201 @@ contract CallServiceTest is Test {
...
+
```
Comment (AntonAndell) [inline contracts/evm/xcall/test/CallService.t.sol:308]: :point_up:

**Review by AntonAndell — CHANGES_REQUESTED** [review]

**Comment by sdpisreddevil** [comment]
@AntonAndell fixed

**Review by AntonAndell — APPROVED** [review]
Okey for now

**Pull request merged**